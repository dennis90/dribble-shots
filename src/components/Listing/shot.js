import React from 'react'

import ShotFooter from './shot-footer'


const Shot = ({shot}) => {
  const getStyle = () => ({
      backgroundImage: `url(${shot.images.normal})`
  })

  return (
    <div className="shot" style={getStyle()}>
      <ShotFooter shot={shot} />
    </div>
  )
}

export default Shot
