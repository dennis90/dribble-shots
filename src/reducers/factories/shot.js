import {Factory} from 'rosie'

export const ShotFactory = new Factory()
  .sequence('id')
  .sequence('title', index => `Shot #${index}`)
  .sequence('description', index => `Shot Description #${index}`)
  .attrs({
    images: {
      hidpi: 'https://dribbble/images/hidpi',
      normal: 'https://dribbble/images/normal',
    },
    views_count: 100,
    likes_count: 100,
  })
  .attr('user', ['user'], user => UserFactory.build(user))

export const UserFactory = new Factory()
  .sequence('name', index => `User #${index}`)
  .attrs({
    avatar_url: 'https://dribbble.com/user/avatar.png',
    html_url: 'https://dribbble.com/user/',
  })
