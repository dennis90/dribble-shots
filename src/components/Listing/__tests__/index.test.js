import React from 'react'
import {shallow} from 'enzyme'
import {OrderedMap} from 'immutable'

import ShotList from '../index'
import {ShotFactory} from '../../../reducers/factories/shot'

describe('ShotList Component', () => {
  it('should render Header, Details and Footer Component', () => {
    const props = ShotFactory.buildList(3).reduce((list, shot) => {
      return list.set(shot.id, shot)
    }, OrderedMap())
    const wrapper = shallow(<ShotList shots={props} />)
    expect(wrapper).toMatchSnapshot('Render a listing with 3 shots')
  })

  it('should return null if no `shots` are provided', () => {
    const wrapper = shallow(<ShotList />)
    expect(wrapper).toMatchSnapshot('Component null mounting')
  })
})
