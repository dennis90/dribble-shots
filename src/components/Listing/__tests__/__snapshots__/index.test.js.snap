// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component null mounting 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/:id(\\\\d+)"
          render={[Function]}
/>,
        undefined,
      ],
      "className": "shots",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/:id(\\\\d+)",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/:id(\\\\d+)"
            render={[Function]}
/>,
          undefined,
        ],
        "className": "shots",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/:id(\\\\d+)",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render a listing with 3 shots 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    shots={
        Immutable.OrderedMap {
            1: Object {
                "description": "Shot Description #1",
                "id": 1,
                "images": Object {
                  "hidpi": "https://dribbble/images/hidpi",
                  "normal": "https://dribbble/images/normal",
                },
                "likes_count": 100,
                "title": "Shot #1",
                "user": Object {
                  "avatar_url": "https://dribbble.com/user/avatar.png",
                  "html_url": "https://dribbble.com/user/",
                  "name": "User #1",
                },
                "views_count": 100,
              },
            2: Object {
                "description": "Shot Description #2",
                "id": 2,
                "images": Object {
                  "hidpi": "https://dribbble/images/hidpi",
                  "normal": "https://dribbble/images/normal",
                },
                "likes_count": 100,
                "title": "Shot #2",
                "user": Object {
                  "avatar_url": "https://dribbble.com/user/avatar.png",
                  "html_url": "https://dribbble.com/user/",
                  "name": "User #2",
                },
                "views_count": 100,
              },
            3: Object {
                "description": "Shot Description #3",
                "id": 3,
                "images": Object {
                  "hidpi": "https://dribbble/images/hidpi",
                  "normal": "https://dribbble/images/normal",
                },
                "likes_count": 100,
                "title": "Shot #3",
                "user": Object {
                  "avatar_url": "https://dribbble.com/user/avatar.png",
                  "html_url": "https://dribbble.com/user/",
                  "name": "User #3",
                },
                "views_count": 100,
              },
        }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/:id(\\\\d+)"
          render={[Function]}
/>,
        Array [
          <withRouter(Shot)
            shot={
                        Object {
                                    "description": "Shot Description #1",
                                    "id": 1,
                                    "images": Object {
                                      "hidpi": "https://dribbble/images/hidpi",
                                      "normal": "https://dribbble/images/normal",
                                    },
                                    "likes_count": 100,
                                    "title": "Shot #1",
                                    "user": Object {
                                      "avatar_url": "https://dribbble.com/user/avatar.png",
                                      "html_url": "https://dribbble.com/user/",
                                      "name": "User #1",
                                    },
                                    "views_count": 100,
                                  }
            }
/>,
          <withRouter(Shot)
            shot={
                        Object {
                                    "description": "Shot Description #2",
                                    "id": 2,
                                    "images": Object {
                                      "hidpi": "https://dribbble/images/hidpi",
                                      "normal": "https://dribbble/images/normal",
                                    },
                                    "likes_count": 100,
                                    "title": "Shot #2",
                                    "user": Object {
                                      "avatar_url": "https://dribbble.com/user/avatar.png",
                                      "html_url": "https://dribbble.com/user/",
                                      "name": "User #2",
                                    },
                                    "views_count": 100,
                                  }
            }
/>,
          <withRouter(Shot)
            shot={
                        Object {
                                    "description": "Shot Description #3",
                                    "id": 3,
                                    "images": Object {
                                      "hidpi": "https://dribbble/images/hidpi",
                                      "normal": "https://dribbble/images/normal",
                                    },
                                    "likes_count": 100,
                                    "title": "Shot #3",
                                    "user": Object {
                                      "avatar_url": "https://dribbble.com/user/avatar.png",
                                      "html_url": "https://dribbble.com/user/",
                                      "name": "User #3",
                                    },
                                    "views_count": 100,
                                  }
            }
/>,
        ],
      ],
      "className": "shots",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/:id(\\\\d+)",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "shot": Object {
            "description": "Shot Description #1",
            "id": 1,
            "images": Object {
              "hidpi": "https://dribbble/images/hidpi",
              "normal": "https://dribbble/images/normal",
            },
            "likes_count": 100,
            "title": "Shot #1",
            "user": Object {
              "avatar_url": "https://dribbble.com/user/avatar.png",
              "html_url": "https://dribbble.com/user/",
              "name": "User #1",
            },
            "views_count": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "shot": Object {
            "description": "Shot Description #2",
            "id": 2,
            "images": Object {
              "hidpi": "https://dribbble/images/hidpi",
              "normal": "https://dribbble/images/normal",
            },
            "likes_count": 100,
            "title": "Shot #2",
            "user": Object {
              "avatar_url": "https://dribbble.com/user/avatar.png",
              "html_url": "https://dribbble.com/user/",
              "name": "User #2",
            },
            "views_count": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "shot": Object {
            "description": "Shot Description #3",
            "id": 3,
            "images": Object {
              "hidpi": "https://dribbble/images/hidpi",
              "normal": "https://dribbble/images/normal",
            },
            "likes_count": 100,
            "title": "Shot #3",
            "user": Object {
              "avatar_url": "https://dribbble.com/user/avatar.png",
              "html_url": "https://dribbble.com/user/",
              "name": "User #3",
            },
            "views_count": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/:id(\\\\d+)"
            render={[Function]}
/>,
          Array [
            <withRouter(Shot)
              shot={
                            Object {
                                          "description": "Shot Description #1",
                                          "id": 1,
                                          "images": Object {
                                            "hidpi": "https://dribbble/images/hidpi",
                                            "normal": "https://dribbble/images/normal",
                                          },
                                          "likes_count": 100,
                                          "title": "Shot #1",
                                          "user": Object {
                                            "avatar_url": "https://dribbble.com/user/avatar.png",
                                            "html_url": "https://dribbble.com/user/",
                                            "name": "User #1",
                                          },
                                          "views_count": 100,
                                        }
              }
/>,
            <withRouter(Shot)
              shot={
                            Object {
                                          "description": "Shot Description #2",
                                          "id": 2,
                                          "images": Object {
                                            "hidpi": "https://dribbble/images/hidpi",
                                            "normal": "https://dribbble/images/normal",
                                          },
                                          "likes_count": 100,
                                          "title": "Shot #2",
                                          "user": Object {
                                            "avatar_url": "https://dribbble.com/user/avatar.png",
                                            "html_url": "https://dribbble.com/user/",
                                            "name": "User #2",
                                          },
                                          "views_count": 100,
                                        }
              }
/>,
            <withRouter(Shot)
              shot={
                            Object {
                                          "description": "Shot Description #3",
                                          "id": 3,
                                          "images": Object {
                                            "hidpi": "https://dribbble/images/hidpi",
                                            "normal": "https://dribbble/images/normal",
                                          },
                                          "likes_count": 100,
                                          "title": "Shot #3",
                                          "user": Object {
                                            "avatar_url": "https://dribbble.com/user/avatar.png",
                                            "html_url": "https://dribbble.com/user/",
                                            "name": "User #3",
                                          },
                                          "views_count": 100,
                                        }
              }
/>,
          ],
        ],
        "className": "shots",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/:id(\\\\d+)",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "shot": Object {
              "description": "Shot Description #1",
              "id": 1,
              "images": Object {
                "hidpi": "https://dribbble/images/hidpi",
                "normal": "https://dribbble/images/normal",
              },
              "likes_count": 100,
              "title": "Shot #1",
              "user": Object {
                "avatar_url": "https://dribbble.com/user/avatar.png",
                "html_url": "https://dribbble.com/user/",
                "name": "User #1",
              },
              "views_count": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "shot": Object {
              "description": "Shot Description #2",
              "id": 2,
              "images": Object {
                "hidpi": "https://dribbble/images/hidpi",
                "normal": "https://dribbble/images/normal",
              },
              "likes_count": 100,
              "title": "Shot #2",
              "user": Object {
                "avatar_url": "https://dribbble.com/user/avatar.png",
                "html_url": "https://dribbble.com/user/",
                "name": "User #2",
              },
              "views_count": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "shot": Object {
              "description": "Shot Description #3",
              "id": 3,
              "images": Object {
                "hidpi": "https://dribbble/images/hidpi",
                "normal": "https://dribbble/images/normal",
              },
              "likes_count": 100,
              "title": "Shot #3",
              "user": Object {
                "avatar_url": "https://dribbble.com/user/avatar.png",
                "html_url": "https://dribbble.com/user/",
                "name": "User #3",
              },
              "views_count": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
